kernel/src/process_load.rs

```diff
@@ -0,0 +1,791 @@
+    /// parse, and just pass through the parse error.
+    fn from(error: tock_tbf::types::TbfParseError) -> Self {
+        ProcessLoadError::TbfHeaderParseFailure(error)
+    }
+}
+
+impl fmt::Debug for ProcessLoadError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            ProcessLoadError::TbfHeaderNotFound => {
+                write!(f, "Could not find TBF header")
+            }
+
+            ProcessLoadError::TbfHeaderParseFailure(tbf_parse_error) => {
+                write!(f, "Error parsing TBF header\n")?;
+                write!(f, "{:?}", tbf_parse_error)
+            }
+
+            ProcessLoadError::NotEnoughFlash => {
+                write!(f, "Not enough flash available for app linked list")
+            }
+
+            ProcessLoadError::NotEnoughMemory => {
+                write!(f, "Not able to meet memory requirements requested by apps")
+            }
+
+            ProcessLoadError::MpuInvalidFlashLength => {
+                write!(f, "App flash length not supported by MPU")
+            }
+
+            ProcessLoadError::MemoryAddressMismatch {
+                actual_address,
+                expected_address,
+            } => write!(
+                f,
+                "App memory does not match requested address Actual:{:#x}, Expected:{:#x}",
+                actual_address, expected_address
+            ),
+
+            ProcessLoadError::IncorrectFlashAddress {
+                actual_address,
+                expected_address,
+            } => write!(
+                f,
+                "App flash does not match requested address. Actual:{:#x}, Expected:{:#x}",
+                actual_address, expected_address
+            ),
+
+            ProcessLoadError::IncompatibleKernelVersion { version } => match version {
+                Some((major, minor)) => write!(
+                    f,
+                    "Process is incompatible with the kernel. Running: {}.{}, Requested: {}.{}",
+                    crate::MAJOR,
+                    crate::MINOR,
+                    major,
+                    minor
+                ),
+                None => write!(f, "Process did not provide a TBF kernel version header"),
+            },
+
+            ProcessLoadError::InternalError => write!(f, "Error in kernel. Likely a bug."),
+
+            ProcessLoadError::CredentialsNoAccept => write!(f, "No credentials accepted."),
+
+            ProcessLoadError::CredentialsReject(index) => {
+                write!(f, "Credentials index {} rejected.", index)
+            }
+        }
+    }
+}
+
+/// Load processes (stored as TBF objects in flash) into runnable
+/// process structures stored in the `procs` array. If a `checker` is
+/// passed, this method scans the footers in the TBF for cryptographic
+/// credentials for binary integrity, passing them to `checker` to decide
+/// whether the process has sufficient credentials to run.
+#[inline(always)]
+#[allow(dead_code)]
```

Comment:
Why do we have dead code?

Reply:
In case the platform calls the version that doesn't check credentials.